packages/flutter_tools/lib/src/web/file_generators/flutter_js.dart

```diff
@@ -56,12 +56,53 @@ _flutter.loader = null;
     });
   }
 
+  /**
+   * Handles the creation of a TrustedTypes `policy` that validates URLs based
+   * on an (optional) incoming array of RegExes.
+   */
+  class FlutterTrustedTypesPolicy {
+    /**
+     * Constructs the policy.
+     * @param {[RegExp]} validPatterns the patterns to test URLs
+     * @param {String} policyName the policy name (optional)
+     */
+    constructor(validPatterns, policyName = "flutter-js") {
+      const patterns = validPatterns || [
+        /\.dart\.js$/,
+        /^flutter_service_worker.js$/
+      ];
```

Comment:
Should we also check that these come from the same domain?

Reply:
For now, I think I'd let the browser do its default Cross-Domain checks; I'm not sure we can 100% mandate what domain these should come from (other than maybe `window.location`?)